### PR TITLE
Add support for creating loadbalancer's VIP using Neutron Network ID

### PR DESCRIPTION
**Pull request by hamzazafar**
**What this PR does / why we need it**:
Add support for creating loadbalancer's VIP using Neutron Network ID.

In addition to Subnet ID, Neutron LBaaS v2 API support creation of loadbalancers using Neutron Network ID. A subnet is chosen from network which has available IP addresses. In case of Octavia backend, IPv4 subnets are preferred over IPv6 subnets.

**Special notes for your reviewer**:

The network ID can be specified in cloud-config file or by using the annotation `loadbalancer.openstack.org/network-id`

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by hamzazafar** [comment]
@adisky can you have a look at my PR, thanks.

File: pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go [hunk 1]
```diff
@@ -450,12 +451,14 @@ func (lbaas *LbaasV2) createLoadBalancer(service *corev1.Service, name, clusterN
...
+	} else if lbClass != nil && lbClass.SubnetID != "" {
+		createOpts.VipSubnetID = lbClass.SubnetID
 	} else {
-		if lbClass != nil && lbClass.SubnetID != "" {
-			createOpts.VipSubnetID = lbClass.SubnetID
-		} else {
-			createOpts.VipSubnetID = lbaas.opts.SubnetID
-		}
+		createOpts.VipSubnetID = lbaas.opts.SubnetID
 	}
```
Comment (lingxiankong) [inline pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go]: Logic is not right here, the network_id and subnet_id can both be provided.

Comment (hamzazafar) [reply pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go]: fixed! can you review. Thanks

**Comment by lingxiankong** [comment]
thanks!

/lgtm

**Comment by FengyunPan2** [comment]
It's ok.
/approve

**Pull request merged**